ros2component/test/test_cli_list.py

```diff
@@ -0,0 +1,150 @@
+# Copyright 2019 Open Source Robotics Foundation, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import contextlib
+import unittest
+
+from launch import LaunchDescription
+from launch.actions import ExecuteProcess
+from launch.actions import OpaqueFunction
+
+from launch_ros.actions import Node
+
+import launch_testing
+import launch_testing.asserts
+import launch_testing.markers
+import launch_testing.tools
+import launch_testing_ros.tools
+
+import pytest
+
+from rmw_implementation import get_available_rmw_implementations
+
+LIST_THREE_NODES = """\
+/ComponentManager
+  1  /listener
+  2  /talker
+  3  /talker
+"""
+
+
+# TODO(BMarchi): Opensplice doesn't get along with running any cli command with ExecuteProcess,
+# it just hangs there making `wait_for_timeout` fail. All tests should run fine with opensplice.
+@pytest.mark.rostest
+@launch_testing.parametrize(
+    'rmw_implementation',
+    [v for v in get_available_rmw_implementations() if v != 'rmw_opensplice_cpp'])
+def generate_test_description(rmw_implementation, ready_fn):
+    additional_env = {'RMW_IMPLEMENTATION': rmw_implementation}
+    component_node = Node(
+        package='rclcpp_components', node_executable='component_container', output='screen')
+    listener_command_action = ExecuteProcess(
+        cmd=['ros2', 'component', 'load', '/ComponentManager',
+             'ros2component_fixtures', 'ros2component_fixtures::Listener'],
+        additional_env={
+            'RMW_IMPLEMENTATION': rmw_implementation,
+            'PYTHONUNBUFFERED': '1'
+        },
+        name='ros2component-cli',
+        output='screen',
+        on_exit=[
+            ExecuteProcess(
+                cmd=['ros2', 'component', 'load', '/ComponentManager',
+                     'ros2component_fixtures', 'ros2component_fixtures::Talker'],
+                additional_env={
+                    'RMW_IMPLEMENTATION': rmw_implementation,
+                    'PYTHONUNBUFFERED': '1'
+                },
+                name='ros2component-cli',
+                output='screen',
+                on_exit=[
+                    ExecuteProcess(
```

Comment:
@BMarchi why nesting actions instead of putting one after the other?

Reply:
It seems that not always the listener gets instantiated after the talkers even if I put them one after the other so I need to ensure this order for proper printing.